web/src/pluginManager.js

```diff
@@ -2085,4 +2094,51 @@ export class PluginManager {
   onClose(_plugin, cb) {
     _plugin.onClose(cb);
   }
+
+  async checkPluginUpdate(plugin) {
+    const pconfig = plugin.config;
+    const config = await this.getPluginFromUrl(
+      pconfig.origin,
+      this.available_plugins
+    );
+    if (pconfig.hash) {
+      if (pconfig.hash !== SparkMD5.hash(config.code)) {
+        if (compareVersions(pconfig.version, "<=", config.version)) {
+          plugin.update_available = true;
+        } else {
+          plugin.update_available = false;
+        }
+      } else {
+        plugin.update_available = false;
+      }
+    } else {
+      if (compareVersions(pconfig.version, "<", config.version)) {
+        plugin.update_available = true;
+      } else {
+        plugin.update_available = false;
+      }
+    }
+    this.update_ui_callback();
+  }
+  checkUpdates() {
+    for (let k in this.plugins) {
+      if (this.plugins.hasOwnProperty(k)) {
+        const plugin = this.plugins[k];
+        if (plugin.config.origin) {
+          this.checkPluginUpdate(plugin);
+        } else {
+          const ps = this.available_plugins.filter(p => {
```

Comment:
If we only expect a single match, and we're not interested in the array but the item in the array, maybe `Array.find` is better?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

Reply:
Sounds good, thanks!